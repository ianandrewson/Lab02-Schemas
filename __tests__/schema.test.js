const { Schema } = require('../lib/schema.js');

const schema = new Schema({
    name: {
        type: String,
        required: true
    }, 
    age: {
        type: Number,
        required: true
    },
    weight: {
        type: String
    }
});

const spot = {
    name: 'spot',
    age: 5,
    weight: '20 lbs'
};

const rover = {
    name: 'rover',
    age: '10'
};

const fido = {
    name: 'Fido',
    age: 'hi'
};

const lassie = {
    age: 80
};

describe('Schema class tests', () => {
    it('should return the input object if it fits the schema', () => {
        expect(schema.validate(spot)).toEqual({ name: 'spot', age: 5, weight: '20 lbs' });
        expect(schema.validate(rover)).toEqual({ name: 'rover', age: 10 });
    });
    it('should throw an error if missing required fields or cannot cast a field', () => {
        expect(() => {schema.validate(fido);}).toThrowErrorMatchingSnapshot();
        expect(() => {schema.validate(lassie);}).toThrowError('missing required field');
    });
});

