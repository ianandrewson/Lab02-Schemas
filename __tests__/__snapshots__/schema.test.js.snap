// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema class tests should throw an error if missing required fields or cannot cast a field 1`] = `"Cannot cast >>hi<< to Number"`;
